test/pythonml/test_pre_processing_flavors.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+import importlib
+import os
+import shutil
+import unittest
+from parameterized import parameterized
+import pandas
+import numpy as np
+from fixtures.unittest_baseclass import BaseUnitTest
+
+
+def load_test_train_targets_and_descriptors():
+    """
+    Loads a set of commonly used pickle files from an updated location in the context object
+    """
+    # import and roload settings to update the context object's path dir
+    import settings
+    importlib.reload(settings)
+    train_target = settings.context.load("train_target")
+    train_descriptors = settings.context.load("train_descriptors")
+    test_target = settings.context.load("test_target")
+    test_descriptors = settings.context.load("test_descriptors")
+    return train_target, train_descriptors, test_target, test_descriptors
+
+
+class TestPreProcessingScalerFlavors(BaseUnitTest):
+    """
+    This class performs unittests for the pre_processing scaler flavors.
+    We may only check regression data here, and assume it would work for the other data categories.
+    """
+
+    def min_max_pass_condition(self, data):
+        for col in data.T:
+            self.assertAlmostEqual(1.0, np.amax(col))
+            self.assertAlmostEqual(0.0, np.amin(col))
+
+    def standard_scaler_pass_condition(self, data):
+        column_means = data.mean(axis=0)
+        column_standard_deviations = data.std(axis=0)
+        for column_mean in column_means:
+            self.assertAlmostEqual(0.0, column_mean)
+        for column_standard_deviation in column_standard_deviations:
+            self.assertAlmostEqual(1.0, column_standard_deviation)
+
+    def run_scaler_flavor_test(self, category, flavor):
+        self.set_pickle_fixtures_path_in_context_object(category, 'unscaled_data')
+        shutil.copy(os.path.join(self.asset_path, flavor), flavor)
+        os.system('python ' + flavor)
+        train_target, train_descriptors, test_target, test_descriptors = load_test_train_targets_and_descriptors()
+        # Check the pass conditions for each data loaded
+        for data in [train_target, train_descriptors]:
+            if 'min_max_scaler' in flavor:
+                self.min_max_pass_condition(data)
+            elif 'standardization' in flavor:
+                self.standard_scaler_pass_condition(data)
+
+    params = [
+        ['regression', "pyml:pre_processing:min_max_scaler:sklearn.pyi"],
+        ['regression', "pyml:pre_processing:standardization:sklearn.pyi"],
+    ]
+    @parameterized.expand(params)
+    def test_flavor(self, category, flavor):
+        self.custom_setup(category)
+        self.run_scaler_flavor_test(category, flavor)
+
+
+class TestPreProcessingDroppers(BaseUnitTest):
+    """
+    This class performs unittests for the pre_processing dropper flavors.
+    """
+
+    def remove_duplicates_pass_condition(self, data):
```

Comment:
Similar comment as the preprocessor file - if the tests have different methods to both run them, and to check their results, that's a sign that it might be better to split them out into separate classes